tt_metal/distributed/mesh_device.cpp

```diff
@@ -47,7 +47,13 @@ static std::unordered_map<LogicalCoordinate, PhysicalCoordinate> load_translatio
         if (mapping.size() != 2 || mapping[0].size() != 2 || mapping[1].size() != 4) {
             throw std::runtime_error("Invalid coordinate format in JSON file: " + filename);
         }
-        result.emplace(LogicalCoordinate{mapping[0][0], mapping[0][1]}, PhysicalCoordinate{mapping[1][1], mapping[1][0], mapping[1][2], mapping[1][3]});
+        result.emplace(LogicalCoordinate{mapping[0][0], mapping[0][1]}, PhysicalCoordinate{
```

Comment:
@cfjchu , related to #12719
Why are mesh_configuration .jsons needed, why is this info not directly read from tt_ClusterDescriptor? What am I missing?

Note on the change you're seeing here, I had to update eth_coord_t to support multiple unconnected clusters in umd, for more info see: https://github.com/tenstorrent/tt-umd/pull/306

Reply:
This layer of the stack is a mapping of physical cluster to a logical mesh setup. What is the usage of cluster_id in UMD? Is it flashed to the chip like the (x, y, rack, shelf) info? If so, I would prefer adding the `cluster_id` to the mesh_config yamls, instead of hardcoding a 0 here.

Reply:
Added to config files

Reply:
Yes, it's effectively used as a translation layer to remap available device cluster onto a 2D logical mesh for upper layers of software. It also makes upper software layers transparent to these UMD changes like you've introduced here.